### PR TITLE
removing setTimeout and adding onBlur event on the cell grid

### PR DESCRIPTION
**Pull request by AmeyDagaria007**
Cell value changed to new value entered, unless used "Esc" key
now this behaviour is working fine 
Please Review @origami-z

### CODE REVIEW

File: packages/grid/src/Grid.tsx [hunk 1]
```diff
@@ -453,7 +453,6 @@ export const Grid = function Grid<T>(props: GridProps<T>) {
     rowIdx: number,
     colIdx: number
   ) => {
-    setTimeout(() => {
```
Comment (ZarkoPernar) [inline packages/grid/src/Grid.tsx:456]: What is the reason you are removing this setTimeout? It was originally put here for a reason, to counter a race condition. I forgot the exact scenario but will need to do some manual QA locally to see if anything is broken.

File: packages/grid/src/NumericColumn.tsx [hunk 2]
```diff
@@ -50,6 +50,10 @@ export function NumericCellEditor<T>(props: NumericEditorProps<T>) {
   const onChange: ChangeEventHandler<HTMLInputElement> = (e) => {
     setEditorText(e.target.value);
   };
+  
+  const onBlur: ChangeEventHandler<HTMLInputElement> = (e) => {
```
Comment (ZarkoPernar) [inline packages/grid/src/NumericColumn.tsx:54]: This should be `FocusEventHandler<HTMLInputElement>`

**Review by ZarkoPernar — CHANGES_REQUESTED** [review]
Thanks for the contribution Amey! I need to talk to our designer in case this functionality was intentional. Even if it was, we can still change it if enough people complain. Out of curiosity, are you using the salt data grid in production?

File: packages/grid/src/Grid.tsx (hunk 1 above)
Comment (ZarkoPernar) [reply packages/grid/src/Grid.tsx:456]: Ah, turns out the test caught the scenario that has a race condition.

Comment (feliciackh) [reply packages/grid/src/Grid.tsx:456]: More context: removing `setTimeOut` results in one of the tests for keyboard navigation failing, we believe `setTimeOut` was initially added to counter virtualization lag, i.e. when we focus on a cell that hasn't been rendered yet

**Comment by AmeyDagaria007** [comment]
> Thanks for the contribution Amey! I need to talk to our designer in case this functionality was intentional. Even if it was, we can still change it if enough people complain. Out of curiosity, are you using the salt data grid in production? (Feel free to reach out to me directly on Symphony)

Sorry is there any open slack or is symphoney an internal tool ?

File: packages/grid/src/NumericColumn.tsx (hunk 2 above)
Comment (AmeyDagaria007) [reply packages/grid/src/NumericColumn.tsx:54]: Yeah Sure will Update it Thanks a lot

File: packages/grid/src/Grid.tsx (hunk 1 above)
Comment (AmeyDagaria007) [reply packages/grid/src/Grid.tsx:456]: @ZarkoPernar @feliciackh  Got it . The Reason to remove this was as it pushes it in event queue so it dosent run in sync and due to that onBlur function is not gettin called as it focus after a while .

**Comment by AmeyDagaria007** [comment]
Thanks @ZarkoPernar Sure . if possible can you please give the link for cla ?

**Comment by origami-z** [comment]
@AmeyDagaria007 See contribution guide [here](https://github.com/jpmorganchase/.github/blob/main/CONTRIBUTING.md).

**Pull request closed without merging**